app/controllers/books_controller.rb

```diff
@@ -6,4 +6,24 @@ def index
   def show
     @book = Book.find(params[:id])
   end
+
+  def edit
+    @book = Book.find(params[:id])
+  end
+
+  def update
+    @book = Book.find(params[:id])
+
```

Comment:
As we can see here, we repeatedly find a book by its id. We can use `before_action` to make the code more DRY.
ref: https://dannemanne.com/posts/ruby_on_rails_explained_before_action

```ruby
class BooksController < ApplicationController
  before_action :set_book, only: %i[show edit update]

  # ...

  def show; end

  def edit; end

  def update
    if @book.update(book_params)
      redirect_to @book
    else
      render :edit
    end
  end

  # ...

  private

  def set_book
    @book = Book.find(params[:id])
  end
end
```